Common/Data/UniverseSelection/Universe.cs

```diff
@@ -188,6 +188,11 @@ public virtual Security CreateSecurity(Symbol symbol, IAlgorithm algorithm, Mark
         /// <returns>All subscriptions required by this security</returns>
         public virtual IEnumerable<SubscriptionRequest> GetSubscriptionRequests(Security security, DateTime currentTimeUtc, DateTime maximumEndTimeUtc)
         {
+            if (currentTimeUtc > maximumEndTimeUtc)
```

Comment:
Perhaps more correctly would be to avoid running universe selection at all when this condition is true. One way to accomplish this would be at the top of `ApplyUniverseSelection` to check the current frontier time and the algorithm end time (which gets passed in here as `currentTimeUtc` and `maximumEndTimeUtc` respectively). In this case, we'd still fetch data for universe selection that we won't act on, so maybe it even belongs a level higher... so we *prevent* universe selection from even fetching data to run on when the frontier is past the algorithm end time, This would probably require inspecting the current start/end times of the universe subscriptions to ensure they don't go past the algorithm end date.

Either way we go, we want to make sure that this condition isn't required to be implemented by derived types since this is a `virtual` method.

Reply:
I agree on moving the time check up one level, to the top of `ApplyUniverseSelection`, not sure about moving it higher: in the daily resolution universe case, doesn't the last bar of the backtesting period have to be emitted anyway (`EndTime` is one second after algorithm end time) ?

This is a simple test case algorithm:
```
    public class BasicTemplateAlgorithm : QCAlgorithm
    {
        public override void Initialize()
        {
            SetStartDate(2017, 1, 4);
            SetEndDate(2017, 1, 10);
            UniverseSettings.Resolution = Resolution.Daily;
            AddUniverse(Universe.DollarVolume.Top(2));
        }
    }
```